susemanager-utils/susemanager-sls/src/tests/test_module_mgrnet.py

```diff
@@ -0,0 +1,114 @@
+import sys
+
+from unittest.mock import MagicMock, patch
+from . import mockery
+
+mockery.setup_environment()
+
+from ..modules import mgrnet
+
+
+mgrnet.__salt__ = {}
+mgrnet.__utils__ = {}
+
+
+def test_mgrnet_virtual():
+    """
+    Test __virtual__ function for the possible cases
+    when either 'host' or 'nslookup' is available or none of them
+    """
+
+    with patch.dict(
+        mgrnet.__utils__,
+        {"path.which": MagicMock(side_effect=[True, False, True, False, False])},
+    ):
+        ret = mgrnet.__virtual__()
+        assert ret is True
+
+        ret = mgrnet.__virtual__()
+        assert ret is True
+
+        ret = mgrnet.__virtual__()
+        assert ret[0] is False
+
+
+def test_mgrnet_dns_fqdns():
+    """
+    Test getting possible FQDNs with DNS tools
+    """
+
+    check_calls = {"host": [], "nslookup": []}
+
+    ipv4_addresses = ["10.0.0.1", "172.16.0.1", "192.168.0.1", "10.10.1.1"]
+    ipv6_addresses = ["fd12:3456:789a:1::1", "fd12:abcd:1234:1::1"]
+
+    names = {
+        "10.0.0.1": "host10.example.org",
+        "172.16.0.1": "host172.example.org",
+        "192.168.0.1": "host10.example.org",
+        "fd12:3456:789a:1::1": "ipv6host3456.example.org",
+        "fd12:abcd:1234:1::1": "ipv6hostabcd.example.org",
+    }
+
+    def _cmd_run_host_nslookup(cmd, ignore_retcode=False):
+        """
+        This function is emulating the output of 'host' or 'nslookup'
+        """
+        ip = cmd[1]
+        cmd = cmd[0]
+        check_calls[cmd].append(ip)
+        rc = 0
+        if ":" in ip:
+            # the conversation is not very accurate here, but it's enough for testing
```

Comment:
nitpick: typo in comment
```suggestion
            # the conversion is not very accurate here, but it's enough for testing
```

Reply:
ouch, thanks :+1: